src/main/java/duke/commands/FindCommand.java

```diff
@@ -0,0 +1,60 @@
+package duke.commands;
+
+import duke.DukeException;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+
+/**
+ * A command type that the chatting bot can read.
+ */
+public class FindCommand extends Command {
+
+    private String targetName;
+
+    /**
+     * The constructor of this class.
+     *
+     * @param targetName
+     */
+    public FindCommand(String targetName) {
+        this.targetName = targetName;
+    }
+
+    /**
+     * The method that includes the execution of the command.
+     *
+     * @param list
+     * @param store
+     * @param ui
+     */
+    @Override
+    public void execute(TaskList list, Storage store, Ui ui) throws DukeException {
+        ui.showMessage("Here are the matching tasks in your list:");
```

Comment:
You may put this line below as it may seem strange when the Duke want to show the matching tasks it finds but then it says no tasks in the list.